staging/src/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go

```diff
@@ -75,6 +75,8 @@ type RESTClientGetter interface {
 	ToRawKubeConfigLoader() clientcmd.ClientConfig
 	// ToHTTPClient returns HTTPClient for sharing transport
 	ToHTTPClient() (*http.Client, error)
+	// SetStdinInUse sets a stdinInUse flag to true and resets clients
+	SetStdinInUse()
```

Comment:
Alternatively we could pass a bool here to be more future proof. Thoughts?

Reply:
Personally, I'm not a fan of this approach, can we split that bit out of this PR and handle just the transport share in this PR?

Reply:
- even though this approach is not that pretty, it is a bit hard to come up with an alternative (https://github.com/kubernetes/kubernetes/pull/108459#discussion_r826386790). Maybe We could rename the interface or create additional interface on top of RESTClientGetter?
- spliting the PR is not that easy since we would introduce a regression and could block stdin and not advertise it back to http client

Reply:
> Personally, I'm not a fan of this approach, can we split that bit out of this PR and handle just the transport share in this PR?

+1, I don't think this bit need to escape out of resource.Builder